### DISCUSSION TITLE
Welcome to BasicTS Discussions!

### DISCUSSION THREAD

**Discussion by zezhishao** [GestaltCogTeam/BasicTS#48] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Houshilong1** [comment]
我的电脑只有cpu，我不知道该如何修改experiments中training.py文件中的代码为使用cpu训练

**Reply by zezhishao** [reply]
直接把—gpus 修改为””（删除gpu的数字变成一个空字符串），然后把配置文件修改为CFG.GPU_NUM=0即可。

**Reply by Houshilong1** [reply]
非常感谢，已解决

**Comment by malay-gaherwar** [comment]
I have a few questions about the hyperparameters used in your implementation and how they differ from the original code. As an example I will take the case of STNorm for this disucssion

1) Is the input length same as sequence length?
2) What does output dimension =12 mean here? In 
3) Is there a reason you took n_layers as 2 when the paper took it as 4?

I think it would be ideal if you could add detailed documentation for at least one baseline (in particular the STD-MAE, STNorm, AGCRN or STAEFormer)
Thanks

**Reply by zezhishao** [reply]
Hey, thanks for reaching out with your question.

The STNorm in BasicTS retains its original implementation, but some of its hyperparameters can be somewhat misleading. I’ve added clarifications in `baselines/STNorm/README.md`.

Here’s a breakdown for clarity:
- `in_dim` and `out_dim` are distinct concepts.
- `in_dim` indicates the number of input channels, specifically the `C` in `[B, L, N, C]`.
- In contrast, `out_dim` refers to the output length, corresponding to the `L` in `[B, L, N, C]`.
- Also, STNorm does not inherently specify the input length; instead, it controls the receptive field (i.e., effective input length) by adjusting the number and size of convolutional layers and kernels.

I’ve also updated the configurations for AGCRN and STAEformer. However, I don’t find their parameters as potentially confusing—they generally align well with the descriptions in their respective papers. Could you clarify what specific details you're looking for in the "detailed documentation" of these baselines?

**Comment by Les1ie** [comment]
当前的版本（BasicTS=1.1.0）中要获取图结构是否得自己实现对`adj_mx.pkl`的读取？
